src/main/java/undo/actions/Action.java

```diff
@@ -0,0 +1,43 @@
+package undo.actions;
+
+public interface Action<T> {
+
+    @SuppressWarnings("SpellCheckingInspection")
+    enum ActionType { CHANGELABELS }
+
+    ActionType getType();
+
+    String getDescription();
+
+    T act(T t);
```

Comment:
I guess it was chosen because the method takes in an object which it then `acts` on? @dariusf any preference?